### PR TITLE
Update by query request implementation

### PR DESCRIPTION
**Pull request by lukasz-falda**
AC
---
* UpdateByQuery was implemented

### CODE REVIEW

**Review by mglowala — APPROVED** [review]

File: src/LuigisBoxBundle/Response/UpdateByQuery/UpdateByQueryResponse.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        return $this->jobId;
+    }
+
+    private function getJobIdFromStatusUrl(string $statusUrl): int
+    {
+        parse_str($statusUrl, $parsed);
+        Assert::count($parsed, 1);
+        $jobId = array_values($parsed);
+        Assert::numeric($jobId[0], sprintf('Job id is no numeric value. Got: %s.', $jobId[0]));
```
Comment (malarzm) [inline src/LuigisBoxBundle/Response/UpdateByQuery/UpdateByQueryResponse.php]: "is not numeric" will be enough

File: src/LuigisBoxBundle/Response/UpdateByQuery/UpdateByQueryStatusResponse.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            }
+        }
+    }
+
+    public function getRawResponse(): array
+    {
+        return $this->rawResponse;
+    }
+
+    public function getTrackerId(): string
```
Comment (malarzm) [inline src/LuigisBoxBundle/Response/UpdateByQuery/UpdateByQueryStatusResponse.php:69]: Is `trackerId` the same thing as `jobId` from `UpdateByQueryResponse`?

File: src/LuigisBoxBundle/Service/UpdateByQueryRequest.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * @var UpdateByRequestFactory
+     */
+    private $factory;
+
+    private $statusFactory;
```
Comment (malarzm) [inline src/LuigisBoxBundle/Service/UpdateByQueryRequest.php:35]: Missing `@var`

**Review by malarzm — APPROVED** [review]

File: src/LuigisBoxBundle/Response/UpdateByQuery/UpdateByQueryStatusResponse.php (hunk 2 above)
Comment (lukasz-falda) [reply src/LuigisBoxBundle/Response/UpdateByQuery/UpdateByQueryStatusResponse.php:69]: No, `jobId` point to the currently processing job. TrackerId is not necessary, but the API returns it here. This is the client's public key.

**Pull request merged**